### DISCUSSION TITLE
Using cockpit to create VMs with latest image

### DISCUSSION THREAD

**Discussion by ACodingfreak** [cockpit-project/cockpit#20878] [Q&A] [answered]
Hi All,

Host machine: latest ubuntu + cockpit

Today I use cockpit to spin out new virtual machines for my dev/test activity. Earlier I was doing the same with virt-manager though.
I primarily use ubuntu 22.04 for my VMs and to maintain consistency I created a static image of the same and clone it everytime rather than  fresh install. Once cloned, I try to update/upgrade the image and install necessary modules using a shell script.

How can I spin out a new virtual machine with the latest ubuntu image instead of cloning from a static image ?

**Comment by jelly** [comment]
This sounds like a more general question about VM creation and not cockpit-machines itself. For creating virtual machines you can use `virt-install` and optionally cloud-init to configure it how you want. This is however outside the scope of Cockpit development.

**Reply by ACodingfreak** [reply]
So no plans for cockpit to support custom vm creation from UI?

**Reply by jelly** [reply, marked as answer]
Cockpit supports "automation" but that is limited too setting the root/user password login
